features/update/worker_restart/feature.java

```diff
@@ -0,0 +1,147 @@
+import update.updateutil.UpdateUtil;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+@ActivityInterface
+public interface feature extends Feature, IntWorkflow {
+
+  @ActivityMethod
+  void block();
+
+  @UpdateMethod()
+  int update(int i);
+
+  @SignalMethod
+  void finish();
+
+  class Impl implements feature {
+
+    private boolean doFinish = false;
+    private int counter = 0;
+
+    static Object updateStartedLock = new Object();
+    static Boolean updateStarted = false;
+
+    static Object updateContinueLock = new Object();
+    static Boolean updateContinue = false;
+
+    private static void signalUpdateStarted() {
+      synchronized (updateStartedLock) {
+        updateStarted = true;
+        updateStartedLock.notify();
+      }
+    }
+
+    private static void waitUpdateStarted() {
+      synchronized (updateStartedLock) {
+        while(!updateStarted) {
+          try {
+            updateStartedLock.wait();
+          } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+          }
+        }
+        updateStarted = false;
+      }
+    }
+
+    private static void signalUpdateContinue() {
+      synchronized (updateContinueLock) {
+        updateContinue = true;
+        updateContinueLock.notify();
+      }
+    }
+
+    private static void waitUpdateContinue() {
+      synchronized (updateContinueLock) {
+        while(!updateContinue) {
+          try {
+            updateContinueLock.wait();
+          } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+          }
+        }
+        updateContinue = false;
+      }
+    }
+
+    @Override
+    public void block() {
+      signalUpdateStarted();
+      waitUpdateContinue();
+    }
+
+
+    @Override
+    public int workflow() {
+      Workflow.await(() -> this.doFinish);
+      return counter;
+    }
+
+    @Override
+    public int update(int i) {
+      var activities = activities(feature.class, builder -> builder
+        .setScheduleToCloseTimeout(Duration.ofSeconds(10)));
+
+      activities.block();
+      var tmp = counter;
+      counter += i;
+      return tmp;
+    }
+
+    @Override
+    public void finish() {
+      this.doFinish = true;
+    }
+
+    @Override
+    public Run execute(Runner runner) {
+      String reason = UpdateUtil.CheckServerSupportsUpdate(runner.client);
+      if (!reason.isEmpty()) {
+        runner.Skip(reason);
+      }
+
+      var run = runner.executeSingleParameterlessWorkflow();
+      var stub = runner.client.newWorkflowStub(feature.class, run.execution.getWorkflowId());
+
+      CompletableFuture<Integer> updateResult = CompletableFuture.supplyAsync(() -> stub.update(1));
+      waitUpdateStarted();
+      runner.getWorkerFactory().shutdown();
+      try {
+        Thread.sleep(1000);
+      } catch (InterruptedException e) {
+        throw new RuntimeException(e);
+      }
```

Comment:
Why sleep?

Reply:
Match the Go SDK feature test

Reply:
I think in Go we had the problem of worker not technically being completely stopped when Stop returned. But I don't think we have to do this here or any SDK where shutdown returning successfully guarantees worker completely stopped.

Reply:
That is not true for Java

Reply:
at least according to the docs for `shutdown`